spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/JobGroupUtils.java

```diff
@@ -55,10 +61,10 @@ public static <T> T withJobGroupInfo(
       SparkContext sparkContext, JobGroupInfo info, Supplier<T> supplier) {
     JobGroupInfo previousInfo = getJobGroupInfo(sparkContext);
     try {
-      setJobGroupInfo(sparkContext, info);
+      setJobGroupInfo(sparkContext, info, false);
```

Comment:
Technically this is a behavior change (which I get, is the goal of the PR) the issue is not sure how many people actually desire the current behavior. If it's not a common case to want to override the JobGroupInfo via this method, then perhaps this behavior can simply change. Otherwise, we may need to support preventing overriding JobGroupInfo through a different mechanism (some kind of run time property which defaults to the original override mechanism).

Reply:
This behavior was introduced in #2287, and it seems the original intention was for better Spark UI presentation, if that's the case, I think overriding job desc is enough.